src/Pos/Ssc/GodTossing/Toss/Logic.hs

```diff
@@ -38,7 +39,8 @@ verifyAndApplyGtPayload
     :: (MonadToss m, MonadError TossVerFailure m)
     => Either EpochIndex (MainBlockHeader ssc) -> GtPayload -> m ()
 verifyAndApplyGtPayload eoh payload = do
-    verifyGtPayload eoh payload  -- not necessary for blocks, but just in case
+    inAssertMode $
+        whenRight eoh $ const $ sanityChecksGtPayload eoh payload
```

Comment:
1. We must **always** do this check when `eoh` is `EpochIndex`. So it's a bug.
2. I asked to write a comment which explains what's going on here.